src/qstls.cpp

```diff
@@ -603,19 +608,20 @@ double AdrFixed::integrand1(const double &q, const double &l) const {
 double
 AdrFixed::integrand2(const double &t, const double &y, const double &l) const {
   const double q = itg.getX();
-  if (q == 0 || t == 0 || y == 0) { return 0; };
```

Comment:
Are you sure that we can safely remove this line?

Reply:
There seems to be no singularity for either q = 0 , y = 0 (w in the expression Panagiotis has) that's why I removed it. As for t = 0 singularities for both expressions adr(x,l), adr(x,0) appear only when x = y. I added the corresponding analytical limits in the two new if statements. I also tested it and didn't see any issues.